### PR TITLE
[16.04] backport of model tool dependencies fix

### PR DESCRIPTION
**Pull request by martenson**
xref: https://github.com/galaxyproject/galaxy/pull/2950
ping @nsoranzo @davebx

### CODE REVIEW

File: lib/galaxy/webapps/tool_shed/controllers/repository.py [hunk 1]
```diff
@@ -1713,8 +1713,8 @@ def get_tool_dependencies( self, trans, **kwd ):
         owner = kwd.get( 'owner', None )
         changeset_revision = kwd.get( 'changeset_revision', None )
         repository = suc.get_repository_by_name_and_owner( trans.app, name, owner )
```
Comment (martenson) [inline lib/galaxy/webapps/tool_shed/controllers/repository.py:1715]: I had to change this to `suc.`

Comment (nsoranzo) [reply lib/galaxy/webapps/tool_shed/controllers/repository.py:1715]: That was expected: https://github.com/galaxyproject/galaxy/pull/2950#discussion_r80083956

Comment (martenson) [reply lib/galaxy/webapps/tool_shed/controllers/repository.py:1715]: @nsoranzo that's why I had no problems doing it :)

File: lib/galaxy/webapps/tool_shed/model/__init__.py [hunk 2]
```diff
@@ -244,11 +244,13 @@ def get_repository_dependencies( self, app, changeset, toolshed_url ):
     def get_type_class( self, app ):
         return app.repository_types_registry.get_class_by_label( self.type )
 
-    def get_tool_dependencies( self, changeset_revision ):
+    def get_tool_dependencies( self, app, changeset_revision ):
+        repo = hg.repository( ui.ui(), self.repo_path( app ) )
+        changeset_revision = metadata_util.get_next_downloadable_changeset_revision( self, repo, changeset_revision )
```
Comment (nsoranzo) [inline lib/galaxy/webapps/tool_shed/model/__init__.py]: Actually that's the one that should be changed: `s/metadata_util/suc/`

**Review by nsoranzo — CHANGES_REQUESTED** [review]

File: lib/galaxy/webapps/tool_shed/model/__init__.py (hunk 2 above)
Comment (martenson) [reply lib/galaxy/webapps/tool_shed/model/__init__.py]: interesting, that one was not in conflict...

Comment (nsoranzo) [reply lib/galaxy/webapps/tool_shed/model/__init__.py]: Because these are new lines.

Comment (martenson) [reply lib/galaxy/webapps/tool_shed/model/__init__.py]: good catch

File: lib/galaxy/webapps/tool_shed/controllers/repository.py (hunk 1 above)
Comment (martenson) [reply lib/galaxy/webapps/tool_shed/controllers/repository.py:1715]: heh, seems I had a problem doing it :D

**Comment by nsoranzo** [comment]
The TS test is still failing https://jenkins.galaxyproject.org/job/docker-toolshed/4140/testReport/shed_functional.functional.test_1400_review_migration_stages/TestToolMigrationStages/test_0005_load_migration_stages_page/ , but I think this should be fixed by updating the freebayes version in `test/shed_functional/functional/test_1400_review_migration_stages.py` . Testing it now.

**Comment by nsoranzo** [comment]
Confirmed. @martenson, can you add the following to this PR ? I have to go home.

```
diff --git a/test/shed_functional/functional/test_1400_review_migration_stages.py b/test/shed_functional/functional/test_1400_review_migration_stages.py
index 8499b68..926f30a 100644
--- a/test/shed_functional/functional/test_1400_review_migration_stages.py
+++ b/test/shed_functional/functional/test_1400_review_migration_stages.py
@@ -29,7 +29,7 @@ class TestToolMigrationStages( ShedTwillTestCase ):
                                       'blast_datatypes', 'bwa_wrappers', 'picard', 'lastz',
                                       'lastz_paired_reads', 'bowtie_color_wrappers', 'bowtie_wrappers',
                                       'xy_plot', 'bam_to_sam' ]
-        migrated_tool_dependencies = [ 'emboss', '5.0.0', 'freebayes', '0.9.4_9696d0ce8a962f7bb61c4791be5ce44312b81cf8',
+        migrated_tool_dependencies = [ 'emboss', '5.0.0', 'freebayes', '0.9.4_a46483351fd0196637614121868fb5c386612b55',
                                        'samtools', '0.1.18', 'blast+', '2.2.26+', 'bwa', '0.5.9', 'picard', '1.56.0',
                                        'lastz', '1.02.00', 'bowtie', '0.12.7', 'FreeBayes requires g++', 'ncurses', 'zlib',
                                        'blast.ncbi.nlm.nih.gov', 'fastx_toolkit', '0.0.13', 'samtools', '0.1.16', 'cufflinks',
```

**Comment by martenson** [comment]
thanks @nsoranzo - we discussed the need to do this with @davebx previously but then it vanished from our minds

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Forward merged.

**Pull request merged**